flink-ml-lib/src/test/java/org/apache/flink/ml/classification/OnlineModelSaveLoadTest.java

```diff
@@ -0,0 +1,178 @@
+import org.apache.flink.streaming.api.datastream.DataStream;
+import org.apache.flink.streaming.api.environment.ExecutionCheckpointingOptions;
+import org.apache.flink.streaming.api.environment.StreamExecutionEnvironment;
+import org.apache.flink.streaming.api.functions.sink.filesystem.bucketassigners.BasePathBucketAssigner;
+import org.apache.flink.streaming.api.functions.sink.filesystem.rollingpolicies.OnCheckpointRollingPolicy;
+import org.apache.flink.table.api.DataTypes;
+import org.apache.flink.table.api.Schema;
+import org.apache.flink.table.api.Table;
+import org.apache.flink.table.api.bridge.java.StreamTableEnvironment;
+import org.apache.flink.types.Row;
+
+import org.apache.commons.collections.IteratorUtils;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+
+import java.io.BufferedReader;
+import java.io.InputStreamReader;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+/** Tests online LogisticRegression model save and load. */
+public class OnlineModelSaveLoadTest {
+    @Rule public final TemporaryFolder tempFolder = new TemporaryFolder();
+    private StreamTableEnvironment tEnv;
+    private StreamExecutionEnvironment env;
+    private static final List<Row> modelRows =
+            new ArrayList<>(
+                    Arrays.asList(
+                            Row.of(Vectors.dense(2.0, 4.5, 3.0)),
+                            Row.of(Vectors.dense(2.1, 4.6, 3.1)),
+                            Row.of(Vectors.dense(20.1, 5.6, 3.1)),
+                            Row.of(Vectors.dense(2.1, 4.7, 3.1))));
+    private String tmpPath;
+
+    @Before
+    public void before() {
+        Configuration config = new Configuration();
+        config.set(ExecutionCheckpointingOptions.ENABLE_CHECKPOINTS_AFTER_TASKS_FINISH, true);
+        env = StreamExecutionEnvironment.getExecutionEnvironment(config);
+        env.enableCheckpointing(100);
+        env.setRestartStrategy(RestartStrategies.noRestart());
+        env.setParallelism(1);
+        tEnv = StreamTableEnvironment.create(env);
+        Schema modelSchema =
+                Schema.newBuilder().column("f0", DataTypes.of(DenseVector.class)).build();
+        Table modelDataTable = tEnv.fromDataStream(env.fromCollection(modelRows), modelSchema);
+        try {
+            tmpPath = tempFolder.newFolder().getAbsolutePath();
+            /* Saves online model data to given path (tmpPath). */
+            saveModelData(
+                    LogisticRegressionModelData.getModelDataStream(modelDataTable),
+                    tmpPath,
+                    new LogisticRegressionModelData.ModelDataEncoder());
+            env.execute();
+        } catch (Exception e) {
+            throw new RuntimeException("Saves model data failed!");
+        }
+    }
+
+    @Test
+    public void testSaveAndLoadSingleModel() throws Exception {
```

Comment:
Can we just load a specific version of the model without enumerating all of them? For example, can we just load model data of `version 2`?

Reply:
I think it's all right.

Reply:
Okay. Let's see how others think~